### PR TITLE
Add Morse potential

### PR DESCRIPTION
**Pull request by cpgoodri**
This PR resolves Issue #6 by adding the Morse potential. Specifically, three functions have been added to energy.py:
- `morse`
- `morse_pair`
- `morse_pair_neighbor_list`

which have the expected behavior. The implementation was modeled after the Lennard Jones potential, though there is a slight difference in how `r_onset` and `r_cutoff` are scaled by `sigma`. I am open to a discussion here...

I have tested this myself, but is there a more rigorous testing protocol in place, or anything like that?

### CODE REVIEW

File: jax_md/energy.py [hunk 1]
```diff
@@ -186,6 +186,75 @@ def lennard_jones_neighbor_list(
...
+def morse_pair(
+    displacement_or_metric,
+    species=None, sigma=1.0, epsilon=5.0, alpha=5.0, r_onset=2.0, r_cutoff=2.5):
+  """Convenience wrapper to compute Morse energy over a system."""
+  sigma = np.array(sigma, dtype=f32)
+  epsilon = np.array(epsilon, dtype=f32)
+  alpha = np.array(alpha, dtype=f32)
+  # CPG comment: I don't think this approach makes sense for Morse potentials
+  #   because, unlike for LJ, it is valid to set sigma=0...
+  #r_onset = r_onset * np.max(sigma)
```
Comment (sschoenholz) [inline jax_md/energy.py:217]: Makes sense, but perhaps remove these lines?

Comment (cpgoodri) [reply jax_md/energy.py]: Removed.

**Comment by cpgoodri** [comment]
> This looks great. Can you take a look at `energy_test.py` and add a test or two following soft_sphere and lennard_jones? I'm not totally sure what the best way to test potentials is, but so far I've made sure that e.g. the minimum is at the right placea nd certain known values are correct. Often it is good just to make sure API changes don't break the potentials.

Ok, I added tests to energy_test.py. They all run fine except for `test_morse_small_neighbor_list_energy`, which is based on `test_lennard_jones_small_neighbor_list_energy`, but this is because there was a fix to the neighbor list for small systems that was made after I made the branch for this PR. So once this gets merged they should all work. 

Should I take this test out and put it back in later, or try to merge on my end, or just leave it?

**Comment by sschoenholz** [comment]
Awesome, thanks for all the changes! This looks great.

**Pull request merged**